common/src/leap/soledad/common/couch.py

```diff
@@ -1474,14 +1506,32 @@ def ensure_database(self, dbname):
         databases. But, in case it ever does, by raising an exception here we
         have one more guarantee that no modified client will be able to
         enforce creation of a database when syncing.
+        This can only be done by using a shell script with proper validation
+        and permissions (create_cmd).
 
         :param dbname: The name of the database to ensure.
         :type dbname: str
 
         :raise Unauthorized: Always, because Soledad server is not allowed to
                              create databases.
+
+        :return: The CouchDatabase object and its replica_uid.
+        :rtype: (CouchDatabase, str)
         """
-        raise Unauthorized()
```

Comment:
I don't understand how this works, I guess this was never called from soledad client before this change, but I don't see where it is called now.